DESIGN.md

```diff
@@ -0,0 +1,43 @@
+## Data Flow
+
+```mermaid
+sequenceDiagram
+    %% Driver is either the CLI or server daemon of the tool responsible for calling the subsequent units
+    participant driver
+
+    %% Fetcher is the component responsible for converting a network URL into raw bytes.
+    participant fetcher
+
+    %% Parser is the component responsible for taking the raw bytes from fetcher and turning it into a structured response 
+    %% The structured response can then be given to the driver to decide what to do with it.
+    participant parser
+
+    %% Metrics is an interface responsible for receiving telemetry in a way that the reporter package can implement. 
+    %% Using their specific counters / spans / gauge, etc.
+    participant metrics
+
+    %% Reporter is the underlying specific reporting toolset collector. Ex: OpenTelemetry, Prometheus, etc.
+    participant reporter
+    
+    note right of driver: Resolving a single URL to structured result
+    driver->>fetcher: provide URI
+    fetcher->>metrics: provide request metrics
+    fetcher->>parser: provides response and stream type
+    fetcher->>metrics: provide response metrics
+    parser->>parser: produce stream type structure
+    parser->>fetcher: provide structure
+    fetcher->>driver: provide structure
```

Comment:
Depending on implementation, if `fetching and parsing` formed one unit of work, this would be the sequence.
If we desired to move fetching as it's own unit to limit which servers required a public internet connection, it may require the driver to have a direct connection to the parsing unit instead.

Reply:
I'd probably merge them for now honestly, simplify the first implementation as much as possible